web-app-router/middleware.ts

```diff
@@ -0,0 +1,3 @@
+import { withMiddlewareAuthRequired } from "@auth0/nextjs-auth0/edge";
+export default withMiddlewareAuthRequired();
+export const config = { matcher: ["/teams", "/teams/:path*", "/join"] };
```

Comment:
can we instead use a whitelist approach where only certain routes are allowed without authentication?

Reply:
Just tested, it's possible to make it for the `/login` page for example, but we also will need to whitelist `/` and I have no idea how to do this correctly with middleware matcher